index.js

```diff
@@ -38,9 +58,50 @@ export const once = event => emitter => Future ((rej, res) => {
   return removeListeners;
 });
 
-//# buffer :: ReadableStream a -> Future Error (Array a)
+//. ### Buffer
+
+//# encode :: Charset -> Buffer -> Future Error String
+//.
+//. Given an encoding and a [Buffer][], returns a Future of the result of
+//. encoding said buffer using the given encoding. The Future will reject
+//. with an Error if the encoding is unknown.
+//.
+//. ```js
+//. > encode ('utf8') (Buffer.from ('Hello world!'));
+//. 'Hello world!'
+//. ```
+export const encode = encoding => buffer => (
+  mapRej (e => new Error (e.message))
```

Comment:
This is because `buffer.toString` throws some really weird doubly-wrapped NodeError of TypeError with extra properties, and I couldn't write an assertion against it. So I figured I'd save my users the trouble of dealing with that altogether. :)